### PR TITLE
MVP: local models via ollama

### PR DESCRIPTION
**Pull request by boopthesnoot**
Sorry for the big PR.

Decoupled the logic from the OpenAI Assistants API, I think the preprocessing was sus and we'll need to validate it again.

### CODE REVIEW

File: alphastats/DataSet_Preprocess.py [hunk 1]
```diff
@@ -183,29 +183,41 @@ def _linear_normalization(self, dataframe: pd.DataFrame):
...
+    def _normalization(self, method: str) -> None:
+        """Normalize across samples."""
+        # TODO make both sample and protein normalization available
         if method == "zscore":
             scaler = sklearn.preprocessing.StandardScaler()
+            # normalize samples => for preprocessing
             normalized_array = scaler.fit_transform(
                 self.mat.values.transpose()
             ).transpose()
+            # normalize proteins => for downstream processing
```
Comment (mschwoer) [inline alphastats/DataSet_Preprocess.py:195]: cf. https://github.com/orgs/MannLabs/projects/12/views/1?filterQuery=-last-updated%3A14days&pane=issue&itemId=79587023

Comment (JuliaS92) [reply alphastats/DataSet_Preprocess.py:195]: Well done for resolving this! We can flesh out the details of what we will implement in the ticket.

**Review by JuliaS92 — APPROVED** [review]
LGTM
I will do a more thourough review when I implement the page tests but I don' see anything obvious now.

**Review by mschwoer — APPROVED** [review]
good enough for now, safe enough to try

**Pull request merged**